tools/build-tools/src/test/typeValidator/classValidator.spec.ts

```diff
@@ -49,92 +49,503 @@ describe("Class", () => {
+            const classOld =
+            `
+            class BaseClass<T> extends Promise<T> {}
+            export class TestClass extends BaseClass<string> {}
+            `;
+
+            const classNew =
+            `
+            class BaseClass<T> extends Promise<T> {}
+            export class TestClass extends BaseClass<number> {}
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // structurally identical extends changes with different names are breaking
+        // (this isn't actually breaking but is difficult to handle and errs on the side
+        // of over-bumping)
+        it("changed class extensions that are structurally identical", () => {
+            const classOld =
+            `
+            class BaseClass<T> extends Promise<T> {}
+            export class TestClass extends BaseClass<string> {}
+            `;
+
+            const classNew =
+            `
+            class BaseClass2<T> extends Promise<T> {}
+            export class TestClass extends BaseClass2<string> {}
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // changing implementations is breaking regardless of structural identicality
+        // (this isn't actually breaking but is difficult to handle and errs on the side
+        // of over-bumping)
+        it("changed class implementations that are structurally identical", () => {
+            const classOld =
+            `
+            interface TestInterface {}
+            export class TestClass implements TestInterface {}
+            `;
+
+            const classNew =
+            `
+            interface TestInterface2 {}
+            export class TestClass implements TestInterface2 {}
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // reordering implementations is not breaking
+        it("reorders changed class implementations", () => {
+            const classOld =
+            `
+            interface TestInterface1 {}
+            interface TestInterface2 {}
+            export class TestClass implements TestInterface1, TestInterface2 {}
+            `;
+
+            const classNew =
+            `
+            interface TestInterface1 {}
+            interface TestInterface2 {}
+            export class TestClass implements TestInterface2, TestInterface1 {}
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.none);
+        });
+    });
+
+    describe("type parameters", () => {
+        // adding/removing class type parameters is not a breaking change if they
+        // aren't also exposed on the public API elsewhere
```

Comment:
Why is this?

Reply:
If there was an (unused in the public API) type param, and then it's removed, wouldn't calling code need to be updated to not include the type param, even if it wasn't used in the public API?

Reply:
yup this test is wrong.  i'll correct the test for this change but the actual fix to correct type parameter handling should happen separately